### ISSUE TITLE
Doesn't seem to convert? 2.3 to 1.4

### ISSUE THREAD

**Issue by NameLord** [YAL-GameMaker-Tools/GMVitalizer#1]
![image](https://user-images.githubusercontent.com/27507408/221412796-60834b29-614d-43a3-a6f1-7d5a87418c48.png)

**Comment by YellowAfterlife** [comment]
The error suggests that the provided project path is incorrect - you could try copying it from from shift+right-click menu

However, as the message says, the tool can't convert GMS 2.3/GM 2022+ projects - the JSON format is different to that of GMS 2.1.x/2.2.x (which the tool was made for).

If you have a budget, I could make a separate tool that converts a GMS2.3 project to 2.2 format (which this tool could then convert to 1.4), but it would be fairly limited as there's no good way to convert most new features into something that'd run on 2.2 or 1.4.

As for alternatives, I can only suggest using GMEdit to copy code from a 2.3 to a 1.4 project faster - it lets you edit the whole object at once, and the `#args` shorthand syntax can help with copying functions.

**Comment by NameLord** [comment]
But why it says partially support 2.3 ?

**Comment by YellowAfterlife** [comment]
There is a second tool in this repository called GMCompFix, which is kind of the opposite of GMVitalizer but not quite - it takes a GMS2 project that was recently converted from GMS1 and replaces some of the "compatibility scripts" by slightly better snippets of code - for example, `__view_get( e__VW.XView, 1 )` becomes `camera_get_view_x(view_camera[1])`. Its manipulations are much simpler than the GMVitalizer's (since it just looks for code to modify, really) and at some point someone paid me to update it.

**Comment by NameLord** [comment]
Wait so these stuff your'e not doiing as a hobby?

**Comment by YellowAfterlife** [comment]
In general, I'm able to release tools and assets for free if it's a small amount of work that has broad appeal (like most of my tutorials, or AseSync), if I already made the thing for my own projects and feel like it will not have high maintenance costs (like GmxGen, sfGML, or GmlCppExtFuncs), or if someone is willing to pay for my time. Time/cost management is essential when you always have half a dozen work projects and people continue to email you questions about your work from _2013_.

In other words, I don't usually make projects like GMVitalizer "at my own expense" because the time would be better spent other things.

**Comment by NameLord** [comment]
So there's no way to convert from 2.3 to 2.2.5 or 1.4 ?
i've saw the GMCompFix but how do i run without
i had my project converted 4 times
1.4 > 2.2.5 > 2.3 > 2.3 Beta
and from there i im stuck can't seem to get back.. because i wanted to use UndertaleModTool to make mod compatibility so other people can mod my game.

**Comment by YellowAfterlife** [comment]
UndertaleModTool has [ongoing efforts](https://github.com/krzys-h/UndertaleModTool/pull/1206) to support current GameMaker versions - chances are that it'll work just fine by the time you finish your game.

**Comment by NameLord** [comment]
Soo about the GMCompFix 
or any way to converting from 2.3 to older versions ?
without having to import one by one or idk...

**Comment by YellowAfterlife** [comment]
No, as I've already said - GMCompFix doesn't convert projects, it fixes up scripts in projects imported into newer GM versions

> There is a second tool in this repository called GMCompFix, which is kind of the opposite of GMVitalizer but not quite - it takes a GMS2 project that was recently converted from GMS1 and replaces some of the "compatibility scripts" by slightly better snippets of code - for example, __view_get( e__VW.XView, 1 ) becomes camera_get_view_x(view_camera[1]). Its manipulations are much simpler than the GMVitalizer's (since it just looks for code to modify, really) and at some point someone paid me to update it.

**Comment by NameLord** [comment]
So the only thing is to just Drag and Drop to an 1.4 Project and then import or add existing from there or ?

**Comment by NameLord** [comment]
or to an 2.2.5 project

**Comment by NameLord** [comment]
My 2.3 project that was converted from 2.2.5 doesn't use any of 2.3 features and neither new stuff that was introduced on that version.

**Comment by YellowAfterlife** [comment]
Either you do it by hand or you write a tool to do so, whichever the quicker (and if you have the experience to do the latter). Technically you can write such a tool in GameMaker itself with `json_parse` and `json_stringify` - just note that it's probably a few dozen hours of work.

**Comment by NameLord** [comment]
Thank you so much for the time that you got for answering my questions.
Can you close this issue since its Finished ?

**Comment by YellowAfterlife** [comment]
I'll close it after I update the README to be clearer about version support.

And finally, if you have updated from 2.2.5 to 2.3 only recently, you could import the 2.2.5 version to 2.3 again, save it in a new directory, and use Beyond Compare (or any other diff tool, really) to compare the two and figure out what changes you need to make in 2.2.5 version before converting it back to 1.4.
